docarray/typing/abstract_type.py

```diff
@@ -0,0 +1,30 @@
+from abc import abstractmethod
+from typing import Any, Type, TypeVar, Union
+
+from pydantic import BaseConfig
+from pydantic.fields import ModelField
+
+from docarray.document import BaseNode
```

Comment:
```suggestion
from docarray.document.base_node import BaseNode
```

This will fix the circular import problem. Otherwise you need to import Document at this level and this create the problem